pkg/engine/engine.go

```diff
@@ -3,43 +3,54 @@ package engine
 import (
 	"errors"
 	"fmt"
+	"path/filepath"
 	"strings"
 	"text/template"
 )
 
 type Engine struct {
 }
 
+const TEMPLATE_EXT = ".tmpl"
+
 func (e Engine) Render(templates map[string][]byte, values map[string]interface{}) (map[string][]byte, error) {
 	t := template.New("gotpl")
 	t.Funcs(funcMap())
 
 	rendered := make(map[string][]byte)
 
 	for name, data := range templates {
-		_, err := t.New(name).Parse(string(data))
-		if err != nil {
-			// TODO: Inner error message includes prefix "template: ", which does not good when printing this error
-			return nil, fmt.Errorf("failed to parse template: %w", err)
-		}
-
-		var buf strings.Builder
-		if err := t.ExecuteTemplate(&buf, name, values); err != nil {
-			// TODO: Inner error message includes prefix "template: ", which does not good when printing this error
-			return nil, fmt.Errorf("failed to execute template: %w", err)
-		}
-
-		output := buf.String()
-
-		// If template uses variables which were undefined, gotpl will insert "<no value>"
-		if strings.Contains(output, "<no value>") {
-			// TODO: Find out which variable was not defined
-			return nil, errors.New("some of the variables used in the template were undefined")
+		if filepath.Ext(name) == TEMPLATE_EXT {
```

Comment:
I'm thinking is skipping non-`.tmpl` files feature of a engine, or should it be same for all engines? So when calling `recipe.engine.Render()` method, should only the templates be passed as a parameter, or all files